### PR TITLE
Add PodSandboxStatus api implementation

### PR DESCRIPTION
**Pull request by xlgao-zju**
A part of #8 

Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/16)

### CODE REVIEW

**Comment by feiskyer** [comment]
@xlgao-zju `PodSandboxStatus` depends on the implementation details of `CreatePodSandbox`. Please wait a while for it.

**Comment by xlgao-zju** [comment]
@feiskyer Ok.

**Comment by xlgao-zju** [comment]
@feiskyer rebased

File: pkg/hyper/hyper.go [hunk 1]
```diff
@@ -112,7 +112,28 @@ func (h *Runtime) DeletePodSandbox(podSandboxID string) error {
...
+
+	state := toPodSandboxState(info.Status.Phase)
+	podIP := ""
+	if len(info.Status.PodIP) > 0 {
+		podIP = info.Status.PodIP[0]
+	}
+
+	podStatus := &kubeapi.PodSandboxStatus{
+		Id:        &podSandboxID,
+		Metadata:  &kubeapi.PodSandboxMetadata{Name: &info.PodName},
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: Need to parse sandbox and fill all members of `PodSandboxMetadata`.

File: pkg/hyper/hyper.go [hunk 2]
```diff
@@ -112,7 +112,28 @@ func (h *Runtime) DeletePodSandbox(podSandboxID string) error {
...
+		State:     &state,
+		Network:   &kubeapi.PodSandboxNetworkStatus{Ip: &podIP},
+		CreatedAt: &info.CreatedAt,
+		Labels:    info.Spec.Labels,
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: Here same. Get and set annotations.

**Comment by xlgao-zju** [comment]
@feiskyer updated

File: pkg/hyper/hyper.go [hunk 3]
```diff
@@ -112,7 +112,42 @@ func (h *Runtime) DeletePodSandbox(podSandboxID string) error {
...
+		Attempt:   &attempt,
+	}
+
+	podStatus := &kubeapi.PodSandboxStatus{
+		Id:          &podSandboxID,
+		Metadata:    podSandboxMetadata,
+		State:       &state,
+		Network:     &kubeapi.PodSandboxNetworkStatus{Ip: &podIP},
+		CreatedAt:   &info.CreatedAt,
+		Labels:      info.Spec.Labels,
```
Comment (feiskyer) [inline pkg/hyper/hyper.go]: nit: fraktiAnnotationLabel should be removed from  info.Spec.Labels:

```go
annotations := getAnnotationsFromLabels(info.Spec.Labels)
delete(info.Spec.Labels, info.Spec.Labels)
```

Comment (xlgao-zju) [inline pkg/hyper/hyper.go]: @feiskyer Thank you for your nit.

**Comment by xlgao-zju** [comment]
@feiskyer updated

**Comment by feiskyer** [comment]
LGTM. Thanks.

**Pull request merged**